ffi/dotnet-idsdk/IoticsIdLib/src/Tools.cs

```diff
@@ -0,0 +1,116 @@
+namespace IOTICS;
+
+class Tools {
+    
+    
+    public static string InvokeGoFunction(Func<IdLib.Return> goFunction)
+    {
+        string? error;
+        IdLib.Return result;
+
+        try
+        {
+            // Call the Go function
+            result = goFunction.Invoke();
+        }
+        catch (Exception ex)
+        {
+            // Handle any unexpected exceptions
+            throw new InvalidOperationException($"Exception when invoking method: {ex.Message}");
+        }
+
+        // Check for errors in r1 and throw an exception if necessary
+        error = IdLib.PtrToStringAndFree(result.r1);
+        if (error != null)
+        {
+            throw new IoticsLibException(error);
+        }
+
+        // Convert r0 to a string and free memory
+        string? value = IdLib.PtrToStringAndFree(result.r0);
+
+        if (value == null)
+        {
+            throw new IoticsLibException("unexpected null result");
+        }
+        return value;
+    }
+    
+    public static string? InvokeGoFunction(Func<IntPtr> goFunction)
+    {
+        IntPtr result;
+
+        try
+        {
+            // Call the Go function
+            result = goFunction.Invoke();
+        }
+        catch (Exception ex)
+        {
+            // Handle any unexpected exceptions
+            throw new InvalidOperationException($"Exception when invoking method: {ex.Message}");
+        }
+
+        // Handle the case where the Go function returns nil
+        if (result == IntPtr.Zero)
+        {
+            return null;
+        }
+
+        // Convert r0 to a string and free memory
+        string? value = IdLib.PtrToStringAndFree(result);
+
+        if (value == null)
+        {
+            throw new IoticsLibException("unexpected null result");
+        }
+        return value;
+    }
+
+    // Check if the given string is a URL
+    public static bool IsUrl(string input)
+    {
+        return Uri.TryCreate(input, UriKind.Absolute, out _);
+    }
+
+    // Remove the trailing slash from the string
+    public static string RemoveTrailingSlash(string input)
```

Comment:
Do we really need this? IIRC `//` or `/` shouldn't matter and the resolver should handle it fine anyway.
This again is extra logic in the wrapper which we shouldn't need (and if we do - why not make all of them behave the same by stuffing into the ffi or golang layer?

Reply:
I usually validate input as early as possible - I understand the concern and it's a good idea to validate in the ffi. but it's not bad to do it in the client either

Reply:
> and it's a good idea to validate in the ffi.

I actually mean validate in golang, not FFI (keep ffi & users of FFI thin)